### PR TITLE
Create bar chart

### LINKED ISSUE #23: Create stacked bar chart
Create stacked bar chart for agg crime count over time (month, week, day )

### PR DESCRIPTION
**Pull request by 786213750**
- Bar chat for crime type
- Bar chat for neighborhood type

### CODE REVIEW

**Review by MoNorouzi23 — COMMENTED** [review]
Thanks Thomas,
We have main filters and do not need to define them again, now we have different filters for bar charts and they do not work with other filters!
The second point is about count of crimes! As you can see we have 4580 crimes while the bar chart shows for example more than 50 k just for a neighbourhood!

**Comment by s-voon** [comment]
@MoNorouzi23 we talked to the TA and the plan was to have different filters for the barchart alone for a better flow of the dashboard since the barchart is not affected by both filters at the same time. But I do agree that the numbers seems a bit off given that we have a total of 5000 ish data for all and all values is showing 50k.

File: src/app.py [hunk 1]
```diff
@@ -6,163 +6,368 @@
...
     return fig
 
+
+@app.callback(
+    Output("crime-neighbourhood-bar-chat", "figure"),
+    Input("neighbourhood-dropdown-bar", "value"),
+)
+def update_neighbourhood_bar_chart(crime_neighbourhood):
+    if crime_neighbourhood == "All":
+        filtered_df = hourly_df
```
Comment (s-voon) [inline src/app.py]:
```suggestion
        filtered_df = crime_df
```

File: src/app.py [hunk 2]
```diff
@@ -6,163 +6,368 @@
...
+    else:
+        filtered_df = hourly_df[(hourly_df["NEIGHBOURHOOD"] == crime_neighbourhood)]
```
Comment (s-voon) [inline src/app.py]:
```suggestion
        filtered_df = crime_df[(crime_df["NEIGHBOURHOOD"] == crime_neighbourhood)]
```

File: src/app.py [hunk 3]
```diff
@@ -6,163 +6,368 @@
...
+    return fig
+
+
+@app.callback(
+    Output("crime-type-bar-chat", "figure"),
+    Input("crime-type-dropdown-bar", "value"),
+)
+def update_type_bar_chart(crime_type):
+    if crime_type == "All":
+        filtered_df = hourly_df
```
Comment (s-voon) [inline src/app.py]:
```suggestion
        filtered_df = crime_df
```

File: src/app.py [hunk 4]
```diff
@@ -6,163 +6,368 @@
...
+    else:
+        filtered_df = hourly_df[(hourly_df["TYPE"] == crime_type)]
```
Comment (s-voon) [inline src/app.py]:
```suggestion
        filtered_df = crime_df[(crime_df["TYPE"] == crime_type)]
```

**Review by s-voon — COMMENTED** [review]
Thanks for the PR, the code looks so clean now!! I'm sure it took some time so really appreciated it! I just have a few comments:
- I think the main problem is you are using hourly_df instead of crime_df during aggregation, cause hourly_df resample data and there are rows where count is 0. But you are using the .size() function which is counting rows as 1 crime. I think if you just use crime_df instead of hourly_df, the value should fix itself
- I think your graph wouldnt need a title cause technically, the sentence with the dropdown is the title if that makes sense?
- I think we can also word the title a little bit differently? What do you think abt "Crime Count in each Neighbourhood for "dropdown" Crimetype"and "Crime Count per Crime Type in 'dropdown' Neighbourhood" 
Also, thanks for renaming the crime type to a shorter form and including styling the the graphs!

File: src/app.py (hunk 1 above)
Comment (786213750) [reply src/app.py]: Thanks I changed it in the new commit.

**Review by s-voon — APPROVED** [review]
Thank you for the amazing workkk!

**Review by WaleedMahmood1 — APPROVED** [review]
Thanks for the amazing work! Thanks for incorporating the feedback from the other reviewers too!

The dashboard as a whole looks great now! Aside from this, I am wondering if our functionality will be affected if we linked each of the bar charts to only one of the filters. Right now, we have two filter doing the same thing, and connecting the two bar plots to the respective filter above the map plot, would make our dashboard more efficient. 

This is something that we can discuss next week. The rest is wonderful!!

**Comment by MoNorouzi23** [comment]
I agree with Waleed for this week, it can help to have an efficient version!

**Pull request merged**